src/components/PageReady/index.js

```diff
@@ -0,0 +1,112 @@
+/** @format */
+
+import React from 'react';
+import axios from 'axios';
+import EllipsisLoader from 'components/EllipsisLoader';
+import { interpretErrorMessage } from './formatErrors';
+import type RequestShape from './request';
+
+import defaultRequest from './request';
+import Context from './config/Context';
+
+type PropShape = {
+  predefinedAction?: Object,
+  url?: string,
+  postBody?: Object,
+  method?: string,
+  onData?: Function,
+  customErrorMessage?: string,
+};
+
+class PageReady extends React.Component<PropShape> {
+  static defaultProps = {
+    onData() {
+      return true;
+    },
+    customErrorMessage: 'Failed to load',
+    url: '',
+    postBody: {},
+    method: 'get',
+    predefinedAction: null,
+    requestObject: {
+      ...defaultRequest,
+    },
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      ...defaultRequest,
+      data: null,
+    };
+  }
+
+  state: RequestShape;
+
+  componentDidMount() {
+    this.initRequest();
+  }
+
+  initRequest = (): any => {
+    if (this.props.predefinedAction) {
+      this.props.predefinedAction.dispatch();
+    } else {
+      this.setState({
+        ...defaultRequest,
+        // eslint-disable-next-line
+        fetching: true,
+      });
+
+      const requestParams = [
+        this.props.url,
+        this.props.method === 'post' ? { ...this.props.postBody } : {},
+      ];
+
+      axios[this.props.method.toLowerCase()](...requestParams)
+        .then((res: Object): any => {
+          const { data } = res;
+          if (this.props.onData) {
+            this.props.onData(res);
+          }
+          this.setState({
+            // eslint-disable-next-line
```

Comment:
lets see if we can do without these eslint comments

Reply:
these lint comments are required otherwise `npm run test` will refuse to run :/